Keep n biggest parts of each feature and not layer (#3385)
qgis/QGIS-Documentation: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst

Keep n biggest parts
--------------------

> Takes a polygon layer and creates a new polygon layer in which multipart geometries

Reviewer: ```suggestion
Takes a multipolygon layer and creates a new multipolygon layer in which the smallest polygons
```

Author: I don't want to be nitpicking (I promised I stopped) but no, "**smallest** parts of each multipolygon" are not removed. The first and main idea is to "keep the largest parts". Then, **if** there are smaller parts, they are removed. The logic is to keep the bigger ones. so this is what should be expressed first imo. Not all the features have >n parts...

Reviewer: I agree. Another go...

> have been removed, leaving only up to n largest (in terms of area) parts for each feature.

Reviewer: ```suggestion
have been removed, leaving only the *n* largest (in terms of area) parts for each feature.
```

Reviewer: Will the the resulting geometries also be multipart geometries (with only the n biggest parts kept), or will the resulting layer have up to n (non-multi) geometry features for each feature of the input multipart geometry layer?

Author: > Will the the resulting geometries also be multipart geometries (with only the n biggest parts kept)

In my understanding, it's that. If the original layer has 10 features, you get 10 features. If you ask for 3 parts, then for each feature, you get 1, 2 or 3 parts depending on its original number of parts (reason why the *up to n*).
@nyalldawson ?

Author: OK with the second sentence

.. figure:: img/n_biggest.png
   :align: center

>    Clockwise from left-up: original multipart feature, one, two and three biggest parts to keep

Reviewer: ```suggestion
   Clockwise from left-up: original multipart feature, one, two and three biggest parts kept
```

``To keep`` [number]

>   Choose how many biggest parts have to be kept. If 1 is selected, only the

Reviewer: ```suggestion
  Choose how many parts to keep. If 1 is selected, only the
```

>   biggest part of the feature will be saved.

Reviewer: ```suggestion
  biggest part of the feature will be kept.
```
